daemon/daemon.go

```diff
@@ -1505,6 +1500,36 @@ func (daemon *Daemon) Reload(config *Config) error {
 	if config.IsValueSet("debug") {
 		daemon.configStore.Debug = config.Debug
 	}
+	// If no value is set for max-downloads
+	// we assume it is the default value
+	maxDownloads := defaultMaxDownloads
```

Comment:
Guess the problem is this; if it's no longer set, `maxUploads` is set to the default, so  
`maxDownloads != originalMaxDownloads` is never true.